### PR TITLE
fix: update weave client with formatted project name

### PR DESCRIPTION
**Pull request by gtarpenning**
https://wandb.atlassian.net/browse/WB-20396

### CODE REVIEW

File: weave/legacy/wandb_interface/project_creator.py [hunk 1]
```diff
@@ -72,4 +77,6 @@ def _ensure_project_exists(entity_name: str, project_name: str) -> None:
                 raise UnableToCreateProject(
                     f"Failed to create project {entity_name}/{project_name}"
                 )
-    return
```
Comment (tssweeney) [inline weave/legacy/wandb_interface/project_creator.py:75]: Is there ever a case that we need entity too?

Comment (tssweeney) [reply weave/legacy/wandb_interface/project_creator.py:75]: in the future we might want to get the project id from here too for client-side digest calcs

Comment (tssweeney) [reply weave/legacy/wandb_interface/project_creator.py:75]: I'd recommend making a structured return object that we can extend

Comment (gtarpenning) [reply weave/legacy/wandb_interface/project_creator.py:75]: right, on it

Comment (gtarpenning) [reply weave/legacy/wandb_interface/project_creator.py:75]: @tssweeney to get the entity back we are going to have to make a change in the wandb sdk

Comment (tssweeney) [reply weave/legacy/wandb_interface/project_creator.py:75]: we don't need it right now ... just want to make sure the return type is a class that is structured

File: weave/legacy/wandb_interface/project_creator.py [hunk 2]
```diff
@@ -72,4 +81,5 @@ def _ensure_project_exists(entity_name: str, project_name: str) -> None:
...
+    project_name_maybe: typing.Optional[str] = None if not project else project["name"]
```
Comment (tssweeney) [inline weave/legacy/wandb_interface/project_creator.py:84]: nit: project should never be none due to check above

**Review by tssweeney — APPROVED** [review]

File: weave/trace_server/trace_server_interface.py [hunk 3]
```diff
@@ -565,9 +565,15 @@ class FileContentReadRes(BaseModel):
     content: bytes
 
 
+class EnsureProjectExistsRes(BaseModel):
+    project_name: typing.Optional[str]
```
Comment (tssweeney) [inline weave/trace_server/trace_server_interface.py:569]: this should not be optional

File: weave/weave_client.py [hunk 4]
```diff
@@ -380,7 +380,10 @@ def __init__(
         self.ensure_project_exists = ensure_project_exists
 
         if ensure_project_exists:
-            self.server.ensure_project_exists(entity, project)
+            resp = self.server.ensure_project_exists(entity, project)
+            if resp.project_name:
```
Comment (tssweeney) [inline weave/weave_client.py:384]: this is not needed when project_name is not optional

**Pull request merged**